project_addon/project_addon/doctype/timesheet_record/timesheet_record.js

```diff
@@ -13,9 +13,36 @@ frappe.ui.form.on('Timesheet Record', {
 			frm.set_value("from_time", frappe.datetime.now_datetime());
 			frm.refresh_field("from_time");
 		}
+		
+	},
+
+	project: function(frm) {
+		//apply filter to task field on project field selected
+		cur_frm.set_query("task", () => {
+            return {
+                filters: [
+                    ["Task", "project", "=", cur_frm.doc.project]
+                ]
+            }
+        })
+		cur_frm.refresh_fields(["project", "task"])
 	},
-	
 
+	task: function(frm) {
+		//set project if task is clicked first
+		if(cur_frm.doc.project == null){
+			frappe.db.get_value('Task', cur_frm.doc.task, 'project')
+				.then(r => {
+					let proj = r.message.project;
+					cur_frm.set_value("project",proj);
```

Comment:
This is correct but can be done in fewer lines:
frappe.db.get_value('Task', frm.doc.task, 'project', (r) => {
		        cur_frm.set_value("project",r.project);
		 });